quantities/registry.py

```diff
@@ -1,41 +1,49 @@
 """
 """
 
+import ast
 import re
-import builtins
 
 
 class UnitRegistry:
+    # Note that this structure ensures that UnitRegistry behaves as a singleton
 
     class __Registry:
 
         __shared_state = {}
+        whitelist = (
+            ast.Expression,
+            ast.Constant,
+            ast.Name,
+            ast.Load,
+            ast.BinOp,
+            ast.UnaryOp,
+            ast.operator,
+            ast.unaryop,
+            ast.Num,
+        )
```

Comment:
I think I would need to carefully check this for our whitelist to make sure everything gets through. But this seems like a great idea overall.  Do we want to add a comment to the stackoverflow so we know where the idea came from? This strategy is super cool, but I don't think it would be understandable to a new developer without a comment or link.

Reply:
All of the quantities and Neo core tests pass with this whitelist, but yes, I would like some more eyes on this.

I originally thought adding the link to the SO post in the PR was sufficient, but on reflection I think you're right, it should also be added as a comment in the code.

Reply:
I accidentally deleted my quantities dev env. I'll make a new one and run some tests to see, but neo tests passing makes me feel better! I'll try to read a bit more on the ast stuff and let you know soon if I find anything.

Reply:
well, we could write a formal grammar for these strings, using [pyparsing](https://github.com/pyparsing/pyparsing) for example, but that seems like overkill, and might have a performance impact.

Reply:
No, I've been reflecting more and I think this is fine as is. I think that my example is misuse by the user. I wish we could protect the user, but if they want to do something like this it's on them. We've protected from the security risk so I'm good with this.